### ISSUE TITLE
Don't prefer number properties for Infinity by default

### ISSUE THREAD

**Issue by Richienb** [sindresorhus/eslint-plugin-unicorn#2311]
[We were discussing this on the tc39 forums](https://es.discourse.group/t/math-clamp-strawman/1059/17)... I don't think the consistency is worth it.

`Infinity` and `-Infinity` require less cognitive load and are easier to write than `Number.POSITIVE_INFINTY` and `Number.NEGATIVE_INFINITY`.

**Comment by sindresorhus** [comment]
Yeah, after using the rule for a while, I agree that it does not provide any benefits. I think we can change the default for the `checkInfinity` option in the next version. However, it would be nice then to have a rule to prefer `Infinity` over `Number.POSITIVE_INFINITY`.

**Comment by Richienb** [comment]
For now, we could set `checkInfinity: false`

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/8ca00a46c6667bc9438f3f20b82874796466702e/rules/prefer-number-properties.js#L81

then later, perhaps we replace `checkInfinity` with a new option `preferInfinityProperties` where we explicitly enforce it?

**Comment by sindresorhus** [comment]
It doesn't really fit the rule. The rule description is:

> Prefer `Number` static properties over global ones.

**Comment by Richienb** [comment]
Maybe we could rename the rule from `prefer-number-properties` to just `number-properties`?

**Comment by Richienb** [comment]
Maybe in the future we could also stop preferring `Number.NaN` over `NaN`? Wondering what your thoughts are.
